File: backend/src/main/java/com/trialmaple/model/enums/GameMode.java
package com.trialmaple.model.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum GameMode {
    CLASSIC_TMNF_TRIAL(TmGame.TMNF, TmCategory.TRIAL, null),
    GEOGUESSR_TMNF_TRIAL(TmGame.TMNF, TmCategory.TRIAL, "TMNF_TRIAL"),
    BLUR_TMNF_TRIAL(TmGame.TMNF, TmCategory.TRIAL, "TMNF_TRIAL"),
    ZOOM_TMNF_TRIAL(TmGame.TMNF, TmCategory.TRIAL, "TMNF_TRIAL"),
    CLASSIC_TMNF_RPG(TmGame.TMNF, TmCategory.RPG, null),
    CLASSIC_TM2_TRIAL(TmGame.TM2, TmCategory.TRIAL, null),
    CLASSIC_TM2_RPG(TmGame.TM2, TmCategory.RPG, null),
    CLASSIC_TM2020_TRIAL(TmGame.TM2020, TmCategory.TRIAL, null),
    CLASSIC_TM2020_RPG(TmGame.TM2020, TmCategory.RPG, null),
    GEOGUESSR_TM2020_RPG(TmGame.TM2020, TmCategory.RPG, "TM2020_RPG");

    private final TmGame tmGame;
    private final TmCategory tmCategory;
    private final String picturesFolderName;

}

File: shared/src/types/api/gameMode.ts
export type ClassicGameMode = 'CLASSIC_TMNF_TRIAL'
    | 'CLASSIC_TMNF_RPG'
    | 'CLASSIC_TM2_TRIAL'
    | 'CLASSIC_TM2_RPG'
    | 'CLASSIC_TM2020_TRIAL'
    | 'CLASSIC_TM2020_RPG';

export type GeoguessrGameMode = 'GEOGUESSR_TMNF_TRIAL' | 'GEOGUESSR_TM2020_RPG';

export type BlurGameMode = 'BLUR_TMNF_TRIAL';

export type ZoomGameMode = 'ZOOM_TMNF_TRIAL';

export type GameMode = ClassicGameMode | GeoguessrGameMode | BlurGameMode | ZoomGameMode;

export const GAME_MODE_DISPLAY_NAMES: Record<GameMode, string> = {
    'CLASSIC_TMNF_TRIAL': 'TMNF Trial Classic',
    'CLASSIC_TMNF_RPG': 'TMNF RPG Classic',
    'CLASSIC_TM2_TRIAL': 'TM2 Trial Classic',
    'CLASSIC_TM2_RPG': 'TM2 RPG Classic',
    'CLASSIC_TM2020_TRIAL': 'TM2020 Trial Classic',
    'CLASSIC_TM2020_RPG': 'TM2020 RPG Classic',
    'GEOGUESSR_TMNF_TRIAL': 'TMNF Trial Geoguessr',
    'GEOGUESSR_TM2020_RPG': 'TM2020 RPG Geoguessr',
    'BLUR_TMNF_TRIAL': 'TMNF Trial Blurred',
    'ZOOM_TMNF_TRIAL': 'TMNF Trial Zoomed'
};
